File: examples/order-api/python/src/order_api/notify_client.py
from __future__ import annotations

import logging

import httpx

logger = logging.getLogger("order_api.notify")


def notify_order_created(base_url: str | None, order_id: str) -> None:
    if not base_url or not base_url.strip():
        logger.info("notify skipped: EXTERNAL_HTTP_URL unset", extra={"order_id": order_id})
        return
    url = f"{base_url.rstrip('/')}/notify"
    try:
        with httpx.Client(timeout=2.0) as client:
            r = client.post(url, json={"order_id": order_id})
            logger.info(
                "notify done",
                extra={"order_id": order_id, "status_code": r.status_code, "url": url},
            )
    except httpx.HTTPError as e:
        logger.warning("notify failed", extra={"order_id": order_id, "error": str(e)})

File: examples/order-api/kotlin/bin/main/dev/pykotmig/orderapi/NotifyClient.kt
package dev.pykotmig.orderapi

import io.ktor.client.HttpClient
import io.ktor.client.request.post
import io.ktor.client.request.setBody
import io.ktor.client.statement.bodyAsText
import io.ktor.http.ContentType
import io.ktor.http.contentType
import kotlinx.serialization.Serializable
import org.slf4j.LoggerFactory

class NotifyClient(
    private val http: HttpClient,
    private val config: AppConfig,
) {
    private val log = LoggerFactory.getLogger("order_api.notify")

    suspend fun notifyOrderCreated(orderId: String) {
        val base = config.externalHttpUrl?.trim()?.takeIf { it.isNotEmpty() }
        if (base == null) {
            log.info("notify skipped: EXTERNAL_HTTP_URL unset order_id={}", orderId)
            return
        }
        val url = "${base.trimEnd('/')}/notify"
        try {
            val r =
                http.post(url) {
                    contentType(ContentType.Application.Json)
                    setBody(NotifyBody(orderId))
                }
            log.info("notify done order_id={} status={} url={}", orderId, r.status, url)
        } catch (e: Exception) {
            log.warn("notify failed order_id={} error={}", orderId, e.message)
        }
    }

    @Serializable
    private data class NotifyBody(val order_id: String)
}
